### PR TITLE
feat(web)!: SPA

### PR DESCRIPTION
**Pull request by jrasm91**
This PR removes the `immich-web` and `immich-proxy` containers in favor of (1) a static web build and (2) using expressjs to serve the static web build from `immich-server` directly. Some of the changes required to make this happen include:

- `immich-server` rewrites fallthrough routes to `index.html`, so a refresh on `/photos` sends back `index.html`.
- `immich-server` hosts the api on `/api` instead of `/`
- `immich-server` exposes port 2283 (the port previously used by `immich-proxy`)
- the web project uses the sveltekit static adapter
- the web project no longer has any `page.server.ts` files, which have been moved to `page.ts`
- the web project has disabled `ssr` across the board
- the web project, during dev, reverse proxies to `immich-server`
- the server docker build now uses the root context `.` instead of `server`, since it needs access to the web files for the production build

TODO:
- [x] Update server to send custom.css as an actual stylesheet
- [x] Figure out what to do with (if anything) the .well-known immich stuff

### CODE REVIEW

**Review by alextran1502 — APPROVED** [review]

File: docker/docker-compose.yml
Comment (alextran1502) [inline docker/docker-compose.yml:1]: We forget to expose the port in the server container in this file, correct?

```
    ports:
      - 2283:3001
```

Comment (jrasm91) [reply docker/docker-compose.yml:1]: Looks like it. I'll add it.

File: server/src/immich/controllers/app.controller.ts [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
+
+  @ApiExcludeEndpoint()
+  @Get('.well-known/immich')
+  getImmichWellKnown() {
+    return {
+      api: {
+        endpoint: '/api',
+      },
+    };
+  }
```
Comment (danieldietzler) [inline server/src/immich/controllers/app.controller.ts:18]: What is this endpoint? lol.

**Review by danieldietzler — APPROVED** [review]

File: server/src/immich/controllers/app.controller.ts (hunk 1 above)
Comment (jrasm91) [reply server/src/immich/controllers/app.controller.ts:18]: People don't want to type in /api on the mobile app, so this is a "discovery" endpoint to figure out the location of the API.

Comment (danieldietzler) [reply server/src/immich/controllers/app.controller.ts:18]: Oh I see that makes sense!

**Comment by benmccann** [comment]
Would you consider using `adapter-node` rather than Express + `adapter-static`? 

Using the `fallback` option from SvelteKit is not great for performance because it forces extra network round trips. E.g. if you plug https://demo.immich.app/auth/login into https://pagespeed.web.dev/ then you can see if fails core web vitals.

Using `adapter-node` should be relatively similar to Express from a deployment perspective.

**Comment by jrasm91** [comment]
We actually were using that prior to this pull request.

Maybe if there was a convenient way to run it within an existing express app, otherwise the added deployment complexity for a self hosted app isn't worth the performance impact.

**Comment by benmccann** [comment]
It should be able to run it within an existing express app without too much trouble. Here's an example: https://kit.svelte.dev/docs/adapter-node#custom-server. I'm a maintainer on SvelteKit, so happy to help make sure it's feasible. It's really cool to see Immich using SvelteKit!

Out of curiosity, what else is the express app responsible for? Is it also the API server? I'd be curious to learn about the performance impact you're referring to as we just added docs about performance and have a PR out to update the SPA docs as well. We're somewhat trying to guide users away from SPAs for performance reasons, but want to learn more about how people are using it in the wild to make sure we're considering everything and making the alternatives just as easy when possible.

**Comment by jrasm91** [comment]
The main advantage for the SPA or disadvantage of the node adapter is deployment complexity. I love the simplicity of static builds personally. For a lot of applications the slight increase of the first page load is insignificant compared to the additional deployment complexity of managing another service. This was the case with Immich. 

We used to run svelte kit as a separate container and this introduced a significant amount of complexity, especially for a self hosted application.

 Seeing that it is possible to use the handler directly with our existing express application looks really interesting though. I will have to revisit that and see if that would work for us or not.

**Comment by benmccann** [comment]
Yeah, totally get wanting to minimize the number of containers running! Thanks for the response!

**Pull request merged**